examples/decoding/plot_decoding_rsa.py

```diff
@@ -0,0 +1,143 @@
+It consists in characterizing the structure of the confusion matrix to infer
+the similarity between brain responses and serve as a proxy for characterizing
+the space of mental representations [1-3].
+
+In this example, we perform RSA on spiking data from [4] where the macaque was
+presented to seven classes of pictures located in one of three spatial
+positions.
+
+References
+----------
+[1] Shepard, R. "Multidimensional scaling, tree-fitting, and clustering."
+ Science 210.4468 (1980): 390-398.
+[2] Laakso, A. & Cottrell, G.. "Content and cluster analysis:
+ assessing representational similarity in neural systems." Philosophical
+ psychology 13.1 (2000): 47-76.
+[3] Kriegeskorte, N., Marieke, M., & Bandettini.  P. "Representational
+ similarity analysis-connecting the branches of systems neuroscience."
+ Frontiers in systems neuroscience 2 (2008): 4.
+[4] Cichy, R. M., Pantazis, D., & Oliva, A. "Resolving human object recognition
+in space and time. Nature neuroscience" (2014): 17(3), 455-462
+"""
+# Authors: Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+print(__doc__)
+
+import os.path as op
+import numpy as np
+from pandas import read_csv
+import matplotlib.pyplot as plt
+
+from mne import find_events, Epochs, pick_types
+from mne.io import Raw
+
+from sklearn.model_selection import StratifiedKFold
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.linear_model import LogisticRegression
+from sklearn.metrics import roc_auc_score
+from sklearn.manifold import MDS
+
+data_path = '/media/DATA/Pro/Projects/NewYork/cichy/'
+###############################################################################
+
+# Define stimulus - trigger mapping
+fname = op.join(data_path, 'visual_stimuli.csv')
+conds = read_csv(fname)
+triggers = set(conds['trigger'])
+inv_conds = dict()
+for val in triggers:
+    sub = conds.query('trigger == %i' % val)
+    inv_conds[val] = sub['condition'].iloc[0]
+
+# Read MEG data
+fname = op.join(data_path, 'sample_subject_0_tsss_mc.fif')
+raw = Raw(fname)
+picks = pick_types(raw.info, meg='mag')[::2]
+events = find_events(raw, min_duration=.002)
+
+# Select & Format event values to match ad-hoc categories
+events = np.array([ev for ev in events if ev[2] in set(conds['trigger'])])
+# le = LabelEncoder()
+# events[:, 2] = le.fit_transform([inv_conds[val] for val in events[:, 2]])
+
+# Epoch data
+is_stim = [val in triggers for val in events[:, 2]]
+events = events[np.where(is_stim)[0]]
+epochs = Epochs(raw, events=events, decim=4, baseline=None, picks=picks,
+                tmin=.050, tmax=.800)
+
+##############################################################################
+# Representational Similarity Analysis is a neuroimaging-specific
+# appelation to refer to statistics applied to the confusion matrix.
+
+# Classify on the average ERF
+clf = make_pipeline(StandardScaler(),
+                    LogisticRegression(solver='lbfgs'))
```

Comment:
I like to educational reasons to make explicit model parameters even if we use defaults